core/src/main/java/org/openstack4j/openstack/manila/domain/actions/ServiceActions.java

```diff
@@ -0,0 +1,38 @@
+package org.openstack4j.openstack.manila.domain.actions;
+
+import org.openstack4j.model.ModelEntity;
+
+/**
+ * Actions to force down/force up manila services
+ */
+public class ServiceActions implements ModelEntity {
+    private String binary;
+    private String host;
+    private boolean forced_down;
+
+    private ServiceActions(String binary, String host, boolean forced_down) {
+        this.binary = binary;
+        this.host = host;
+        this.forced_down = forced_down;
+    }
+
+    public static ServiceActions forceUp(String binary, String host) {
+        return new ServiceActions(binary, host, false);
+    }
+
+    public static ServiceActions forceDown(String binary, String host) {
+        return new ServiceActions(binary, host, true);
+    }
+
+    public String getBinary() {
+        return binary;
+    }
+
+    public String getHost() {
+        return host;
+    }
+
+    public boolean isForced_down() {
```

Comment:
I know the codebase is not uniform in this regard, but please stick with camelCase notation. `isForcedDown`.

Reply:
I agree that the name looks weird but I had to name it that way to make the force up and force down operations work in the system.
I know it sounds strange but if the method is named isForcedDown(), the forcedown and forceup operations fail. Here is the explanation:
The curl request for force up operation is shown below:
********
REQ: curl -g -i  -X PUT https://xxx/v2.1/os-services/force-down -H "Accept: application/json" -H "Content-Type: application/json" 
-H "User-Agent: python-novaclient" -H "X-Auth-Token: xxx" -H "X-OpenStack-Nova-API-Version: 2.11" -d '{"host": "xxx-compute", "binary": "compute1", "forced_down": false}'
********
If you look at the data to be posted, you can see that the system expects a mandatory parameter called 'forced_down' which can be set to either true (force-down operation) or false (force-up operation). This variable gets affected by the way we name the method. If the method is named isForcedDown(), the POST request is sent with below data:
 -d '{"host": "xxx-compute", "binary": "compute1", "forcedDown": false}' and it fails saying mandatory parameter 'forced_down' is missing.

Only when we name the method isForced_down(), the POST request is framed correctly with below data:
 -d '{"host": "xxx-compute", "binary": "compute1", "forced_down": false}' and it succeeds.

I have no idea why this happens but it has to be named this way for it to work.

Reply:
@olivergondza kindly review the updated code.

Reply:
`@JsonProperty` should be used to denote the name of the property we want to serialize instead.

Reply:
Yes, I had already tried that annotation, but the issue was still there.
Below is my code in ServicesActions class where isForced_down is defined:
****
 @JsonProperty("forced_down")
 private boolean isForcedDown;
and I renamed method to isForcedDown()
****
Below is the error I get with above code:
*****
Exception in Test Case: (ClientResponseException: Additional properties are not allowed (u'forcedDown' was unexpected))
org.openstack4j.api.exceptions.ClientResponseException: Additional properties are not allowed (u'forcedDown' was unexpected)
	at org.openstack4j.core.transport.HttpExceptionHandler.mapException(HttpExceptionHandler.java:38) ~[classes/:?]
****

Reply:
I am unsure why that is. I can speculate it is because Jackson is ignoring the field (and its annotation when it is serialized) as it is private, and using the unannotated getter. I would try:

- making the field public
- annotating the getter as well

We will decide what is best when we know more.

Reply:
It worked this time, made changes as suggested.